### PR TITLE
add guide to use Sentry with SolidJS

### PR DESCRIPTION
**Pull request by atilafassina**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

This PR adds a recipe to use Sentry browser SDK with a [Solid](https://solidjs.com) app as discussed in [Add SDK for Solid](https://github.com/getsentry/sentry-javascript/discussions/5552#discussioncomment-7708874)

## Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: src/platforms/javascript/guides/solid/manual-setup.mdx [hunk 1]
```diff
@@ -0,0 +1,266 @@
+---
+title: Manual Setup
```
Comment (AbhiPrasad) [inline src/platforms/javascript/guides/solid/manual-setup.mdx:2]: I think we can remove this, not needed for solidjs

Comment (atilafassina) [reply src/platforms/javascript/guides/solid/manual-setup.mdx]: oops. bad copy/paste.
Sorry about that!!

**Review by AbhiPrasad — APPROVED** [review]
LGTM - thanks @atilafassina for contributing!!

**Comment by atilafassina** [comment]
> LGTM - thanks @atilafassina for contributing!!

Thank **you** for the help!! 🥳

**Comment by AbhiPrasad** [comment]
Won't merge just yet - also need a ✅ from docs folks to confirm we're okay with this, but then we can merge. Appreciate the patience in the mean time!

File: src/platform-includes/getting-started-config/javascript.solid.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+To use the SDK, initialize it in your Solid entry point before bootstrapping your app. In a typical Solid project that is the render-client.js, or render-client.tsx.
```
Comment (lizokm) [inline src/platform-includes/getting-started-config/javascript.solid.mdx:1]:
```suggestion
To use the SDK, initialize it in your Solid entry point before bootstrapping your app. (In a typical Solid project that's the render-client.js, or render-client.tsx.)
```

File: src/platform-includes/getting-started-config/javascript.solid.mdx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    // Set tracesSampleRate to 1.0 to capture 100%
+    // of transactions for performance monitoring.
+    // We recommend adjusting this value in production
+    tracesSampleRate: 1.0,
+
+    // Set `tracePropagationTargets` to control for which URLs distributed tracing should be enabled
+    tracePropagationTargets: ["localhost", /^https:\/\/yourserver\.io\/api/],
+
+    // Capture Replay for 10% of all sessions,
+    // plus for 100% of sessions with an error
```
Comment (lizokm) [inline src/platform-includes/getting-started-config/javascript.solid.mdx:26]:
```suggestion
    // Capture Replay for 10% of all sessions,
    // plus 100% of sessions with an error
```

File: src/platform-includes/getting-started-config/javascript.solid.mdx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+const app = document.getElementById("app");
+
+if (!app) throw new Error("No #app element found in the DOM.");
+
+render(() => <App />, app);
+```
+
+Once you've done this, the SDK will automatically capture unhandled errors and promise rejections, and monitor performance in the client.
+
+It is also possible to add Client-Side routing to your app with [Solid-Router](https://docs.solidjs.com/guides/how-to-guides/routing-in-solid/solid-router) without any additional setting. As long as you use the default HTML History API to handle routing.
```
Comment (lizokm) [inline src/platform-includes/getting-started-config/javascript.solid.mdx:41]:
```suggestion
It's also possible to add Client-Side routing to your app with [Solid-Router](https://docs.solidjs.com/guides/how-to-guides/routing-in-solid/solid-router) without changing any additional settings. As long as you use the default HTML History API to handle routing.
```

File: src/platform-includes/getting-started-verify/javascript.solid.mdx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  type="button"
+  onClick={() => {
+    throw new Error("Sentry Frontend Error");
+  }}
+>
+  Throw error
+</button>
+```
+
+This snippet adds a button that throws an error in any Solid component you choose. It is useful for testing. It is recommended to base wrap your app within an Error Boundary:
```
Comment (lizokm) [inline src/platform-includes/getting-started-verify/javascript.solid.mdx:12]:
```suggestion
This snippet adds a button that throws an error in any Solid component you choose. It's useful for testing. It's recommended to base wrap your app within an Error Boundary:
```

**Review by lizokm — APPROVED** [review]
This looks good!

**Pull request merged**